### PR TITLE
Fix KeywordScanner case insensitive search

### PR DESCRIPTION
**Pull request by fbruton**

### CODE REVIEW

File: src/main/java/emissary/util/search/KeywordScanner.java
Comment (drivenflywheel) [inline src/main/java/emissary/util/search/KeywordScanner.java:1]: This seems overly complicated, as a side effect of the existing `static` modifier on the `get256Value` method.  If we remove that modifier, we can collapse to the original `get256Value` method signature with no overloads needed:

```
    private int get256Value(final byte b) {
        return caseSensitive ? Byte.toUnsignedInt(b) : lowercase(Byte.toUnsignedInt(b));
    }
```

That simplication allows us to:
1. revert the added complexity in the `analyze` method (no need to pass `caseSensitive` as a param)
2. remove this redundant block in the `match` method:
```
            if (!this.caseSensitive) {
                currentByte = lowercase(currentByte);
            }
```

**Review by drivenflywheel — DISMISSED** [review]
Some minor simplification requested

File: src/main/java/emissary/util/search/KeywordScanner.java
Comment (fbruton) [reply src/main/java/emissary/util/search/KeywordScanner.java:1]: Thank you. Totally agree with this approach.

**Review by arp-0984 — APPROVED** [review]

**Review by drivenflywheel — APPROVED** [review]
LGTM

**Review by SemperFidelisTyrannosaurus — APPROVED** [review]

**Pull request merged**